mobilecoind/strategies/drain-accounts.py

```diff
@@ -0,0 +1,131 @@
+                        help="Amount less than the balance that we attempt to send")
+    return parser.parse_args()
+
+
+def run_test(stub, amount, monitor_id, dest, max_seconds):
+    resp = stub.GetBalance(
+        mobilecoind_api_pb2.GetBalanceRequest(monitor_id=monitor_id))
+    starting_balance = resp.balance
+    print("Starting balance prior to transfer:", starting_balance)
+    tx_stats = {}
+    sync_start = time.time()
+    wait_for_accounts_sync(stub, [monitor_id], 3)
+    print("Time to sync:", time.time() - sync_start)
+    tx_resp = stub.SendPayment(
+        mobilecoind_api_pb2.SendPaymentRequest(
+            sender_monitor_id=monitor_id,
+            sender_subaddress=0,
+            outlay_list=[
+                mobilecoind_api_pb2.Outlay(
+                    value=amount,
+                    receiver=dest,
+                )
+            ],
+            fee=0,
+            tombstone=0,
+        ))
+
+    tx_stats[0] = {
+        'start': time.time(),
+        'time_delta': None,
+        'tombstone': tx_resp.sender_tx_receipt.tombstone,
+        'block_delta': None,
+        'status': TransferStatus.pending,
+        'receipt': tx_resp,
+    }
+    stats = poll(monitor_id, tx_stats, stub)
+    # FIXME: Move max seconds check inside polling
+    assert tx_stats[0]['time_delta'] < max_seconds, "Did not clear in time"
+    assert tx_stats[0]['status'] == TransferStatus.success, "Transfer did not succeed"
+    return stats
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    print(args)
+
+    stub = connect(args.mobilecoind_host, args.mobilecoind_port)
+    source_accounts = [
+        load_key_and_register("{}/{}".format(args.key_dir, k), stub)
+        for k in sorted(
+            filter(lambda x: x.endswith(".json"), os.listdir(args.key_dir)))
+    ]
+
+    dest_addresses = [
+        with b58_pubfile as open(k, "r"):
+            b58_pubfile.read()
+        for k in sorted(
+            filter(lambda x: x.endswith(".b58pub"), os.listdir(args.key_dir)))
+    ]
+
+    # Go through each account and have all their friends transact to them
+    for i, (src_account, dest) in enumerate(zip(accounts, dest_addresses)):
+        wait_for_accounts_sync(stub, src.monitor_id, 3)
+        # Get starting balance
+        resp = stub.GetBalance(
+            mobilecoind_api_pb2.GetBalanceRequest(monitor_id=account_data.monitor_id))
+        balance = resp.balance
+        print("Starting balance for account", i, ":", resp)
+
+        # Note: due to the transaction fee, we can't assume we have enough funds
+        # to divide equally among all our friends, so add an extra factor.
+        amount = balance - args.fee
+
+        # Create a pool of transfers to all other accounts
+        print("Transferring", amount, "each to", len(accounts), "accounts")
+
+        run_test(stub, amount, src_account, dest, args.max_second)
+        print("Test", i, "succeeded:", stats)
+
+    print("All transfers successful")
```

Comment:
All of this could be done in parallel with asyncio + [async grpcio api](https://grpc.github.io/grpc/python/grpc_asyncio.html), we should consider using that in the future

Reply:
ok cool -- so i think, this is not really meant to be optimized for performance, the test-client.py code is just there to exercise mobilecoind APIs, and this one, is just a small script. If there's something to be gained by making it async then by all means lets do it, but I don't think it serves my purposes right now afaict

Reply:
Agreed!